SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialTraceUniqueId.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/Connection/SpatialTraceUniqueId.h"
+
+#include "Utils/SchemaUtils.h"
+
+using namespace SpatialGDK;
+
+namespace
+{
+void ToHex(uint8 N, TCHAR OutHex[3])
+{
+	const TCHAR* Chars = TEXT("0123456789ABCDEF");
+	OutHex[0] = Chars[(N >> 4) & 0xF];
+	OutHex[1] = Chars[N & 0xF];
+#if !PLATFORM_LITTLE_ENDIAN
+	std::swap(OutHex[0], OutHex[1]);
+#endif
+	OutHex[2] = '\0';
+}
+} // namespace
+
+FString EventTraceUniqueId::GetString() const
+{
+	FString Str;
+	TCHAR Tmp[3];
+	for (int i = 0; i < sizeof(InternalBytes); i++)
+	{
+		ToHex(InternalBytes[i], Tmp);
```

Comment:
Would `FString::Printf(TEXT("%02X"), InternalBytes[i])` not give us the right conversion?

Reply:
it does, just requires a dynamic allocation from FString for 3 TCHARs 😦. Can probably just bake this as 2 wide integers instead though which is far simpler

Reply:
What about `CString::Sprintf`?